bootstrapper/internal/journald/journald.go

```diff
@@ -31,17 +33,35 @@ func NewCollector(ctx context.Context) (*Collector, error) {
 	if cmd.Err != nil {
 		return nil, cmd.Err
 	}
-	return &Collector{cmd}, nil
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+	stderrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, err
+	}
+	collector := Collector{cmd, stdoutPipe, stderrPipe}
+	return &collector, nil
+}
+
+// Pipe returns a pipe to read the systemd logs. This should be read with a bufio Reader.
+func (c *Collector) Pipe() (io.ReadCloser, error) {
```

Comment:
I personally think Pipe is fine here given that this is the underlying OS primitave. From a caller perspective it usually doesn't matter if it's an OS pipe or a Go IO pipe - they are both usually used as io.Reader or io.Writer objects and usually only the latter here is left untouched.

But out of your suggestions I would suggest going with `Start()` otherwise. `StartCollecting()` or `Collect()` would imply to me that this function does something asynchronously or blocks, but it is the caller's responsibility to read from the pipe and consume the data - otherwise this process will stand still on a blocking pipe.

Reply:
I personally think Pipe is fine here given that this is the underlying OS primitive and Go also calls it like that (`StdoutPipe`). Both types of pipes almost have a very similar interface anyway (you can implement both as `io.ReadCloser`), it's only the blocking mechanism that's slightly different between the two given that one comes from the OS.

But out of your suggestions I would suggest going with `Start()` otherwise. `StartCollecting()` or `Collect()` would imply to me that this function does something asynchronously or blocks, but it is the caller's responsibility to read from the pipe and consume the data - otherwise this process will stand still on a blocking pipe.

Reply:
It does something asynchronously: it starts another process and doesn't wait for it to finish.

Reply:
That’s not asynchronous. Pipes are synchronous - they block until you read them.

Reply:
Is there a consensus on the new name of the function yet? I would either leave it as `Pipe` or go with `Start` or `Collect`, whichever the three of you think fits best.

Reply:
I'm personally ok with `Pipe()` but slightly prefer `Start()`, for which we have a majority vote. So if you as the author are also ok with that, I'd say we go with `Start()` for now.

Reply:
Sure :)